cmd/hope/vm/create.go

```diff
@@ -0,0 +1,116 @@
+package vm
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+)
+
+import (
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+import (
+	"github.com/Eagerod/hope/cmd/hope/utils"
+	"github.com/Eagerod/hope/pkg/ssh"
+)
+
+var createCmdVmName string
+var createCmdMemory string
+var createCmdCpu string
+
+func initCreateCmdFlags() {
+	createCmd.Flags().StringVarP(&createCmdVmName, "name", "n", "", "what to name the virtual machine. If left blank, defaults to the hope file name.")
+	createCmd.Flags().StringVarP(&createCmdMemory, "memory", "m", "", "how much memory to given the created VM.")
+	createCmd.Flags().StringVarP(&createCmdCpu, "cpu", "c", "", "how many vCPUs to given the created VM.")
+}
+
+var createCmd = &cobra.Command{
+	Use:   "create",
+	Short: "Creates a VM on the specified host.",
+	Args:  cobra.ExactArgs(2),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		hypervisorName := args[0]
+		vmName := args[1]
+
+		hypervisor, err := utils.GetNode(hypervisorName)
+		if err != nil {
+			return err
+		}
+
+		if !hypervisor.IsHypervisor() {
+			return fmt.Errorf("Node %s is not a hypervisor node; cannot create a VM on it", hypervisorName)
+		}
+
+		vms, err := utils.GetVMs()
+		if err != nil {
+			return err
+		}
+
+		vm, err := utils.VMSpec(vmName)
+		if err != nil {
+			return err
+		}
+
+		vmDir := path.Join(vms.Root, vm.Name)
+
+		log.Debug(fmt.Sprintf("Copying contents of %s for parameter replacement.", vmDir))
+		tempDir, err := utils.ReplaceParametersInDirectoryCopy(vmDir, vm.Parameters)
+		if err != nil {
+			return err
+		}
+		defer os.RemoveAll(tempDir)
+
+		tempPackerJsonPath := path.Join(tempDir, "packer.json")
+		bytes, err := ioutil.ReadFile(tempPackerJsonPath)
+		if err != nil {
+			return err
+		}
+
+		var packerSpec PackerSpec
```

Comment:
Maybe move all of this Packer file reading stuff into the packer pkg? Seems more appropriately placed, and can be beefed up within its own context without making these kinds of operations too complicated.